db/data_migration/20200311133826_make_command_paper_numbers_valid.rb

```diff
@@ -0,0 +1,12 @@
+Attachment
+  .where.not(command_paper_number: nil)
+  .where.not(command_paper_number: "")
+  .each do |attachment|
+  valid_number = attachment.command_paper_number # for example, "CM.123-iv "
+    .tr(". ", "") # remove periods and spaces: "CM123-iv"
+    .gsub(/(\d+)/) { |number| ". #{number}" } # inject period & space before number: "CM. 123-iv"
+    .capitalize # "Cm. 123-iv"
+    .gsub(/Cp\./, "CP") # fix the special case "CP", which should be all caps and no period
+    .gsub(/\d+-(.+)/, &:upcase) # make suffix uppercase: "Cm. 123-IV"
```

Comment:
Are we sure this will match the whole suffix? I'm wondering if we need a `$` or `\z` constraint.

Reply:
It's not necessary but it does help to signify intent better 👍